src/core/client/auth/containers/SignInContainer.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from "react";
+import { BadUserInputError } from "talk-framework/lib/errors";
+
+import SignIn, { SignInForm } from "../components/SignIn";
+import {
+  SetViewMutation,
+  SignInMutation,
+  withSetViewMutation,
+  withSignInMutation,
+} from "../mutations";
+
+interface SignInContainerProps {
+  signIn: SignInMutation;
+  setView: SetViewMutation;
+}
+
+interface SignUpContainerState {
+  error: string | null;
+}
+
+export type View = "SIGN_UP" | "FORGOT_PASSWORD";
+
+class SignInContainer extends Component<
+  SignInContainerProps,
+  SignUpContainerState
+> {
+  public state = { error: null };
```

Comment:
We don't need an error state, final form has support for this: See https://github.com/final-form/final-form#error-any-1

Reply:
This is the error that comes back from the server. Are we speaking about the same thing?

Reply:
What I mean is, final form manages the form state for us including this generic form error, so what we are doing here is reinventing the wheel instead of using final form APIs.